tests/projects/test_project.py

```diff
@@ -86,6 +86,22 @@ def test_sync_functions_with_names_different_than_default(rundb_mock):
     assert project.spec._function_definitions == project_function_definition
 
 
+def test_sync_functions_preserves_existing(rundb_mock):
+    project = mlrun.new_project("project-name", save=False)
+    project.set_function("hub://describe", "describe")
+    project.set_function("hub://auto-trainer", "auto-trainer")
+
+    old_trainer = project.spec._function_objects.pop("auto-trainer")
+    old_describe = project.spec._function_objects["describe"]
+
+    project.sync_functions(always=False)
```

Comment:
For completeness, may be interesting to check the situation where you pass `names` of functions, to verify the same logic holds for specific functions as well. Just note that if you want to test the 1st-pass behavior, make sure to reset the project's `_initialized` property before (as it gets set on `sync_functions`).

Reply:
Good points, Saar!

I improved our test coverage. Please, check if this is what you expected.